scripts/zones/Aht_Urhgan_Whitegate/npcs/Asrahd.lua

```diff
@@ -57,7 +57,7 @@ function onEventFinish(player,csid,option)
             player:addKeyItem(id);
             player:messageSpecial(KEYITEM_OBTAINED,id);
             player:delCurrency("imperial_standing", 1000);
-        elseif (option <= 2049) then -- player bought item
+        elseif (option <= 57345) then -- player bought item
```

Comment:
we also need to stop trusting the option result, see
https://github.com/DarkstarProject/darkstar/blob/e760729999ff81e7b9cfb2fbb7fc3cf620416f35/scripts/zones/Tavnazian_Safehold/npcs/Meret.lua#L99
and
https://github.com/DarkstarProject/darkstar/blob/e760729999ff81e7b9cfb2fbb7fc3cf620416f35/scripts/zones/Tavnazian_Safehold/npcs/Yurim.lua#L124
for existing examples of checking if the option code is legit

Reply:
actually just double check they are able to get the item in finish, because the event update can be cheated before they reach here,

Reply:
I have tested getting the items

Reply:
no clue on bits and shifts guys im a simple guy

Reply:
like this for the local var check?  i tried adding the inital var set in onEventUpdate but it wouldnt set the var

function onEventFinish(player,csid,option)
    if (csid == 630) then
        player:setLocalVar("sanctionItem", option)
        if (option == 0 or option == 16 or option == 32 or option == 48) then -- player chose sanction.
            if (option ~= 0) then
                player:delCurrency("imperial_standing", 100);
            end

            player:delStatusEffect(dsp.effect.SIGIL);
            player:delStatusEffect(dsp.effect.SANCTION);
            player:delStatusEffect(dsp.effect.SIGNET);
            local duration = getSanctionDuration(player);
            local subPower = 0; -- getImperialDefenseStats()
            player:addStatusEffect(dsp.effect.SANCTION,option / 16,0,duration,subPower); -- effect size 1 = regen, 2 = refresh, 3 = food.
            player:messageSpecial(SANCTION);

        elseif (option % 256 == 17) then -- player bought one of the maps
            id = 1862 + (option - 17) / 256;
            player:addKeyItem(id);
            player:messageSpecial(KEYITEM_OBTAINED,id);
            player:delCurrency("imperial_standing", 1000);
        elseif (option == player:getLocalVar("sanctionItem")) then -- player bought item
            item, price = getISPItem(option)
            if (player:getFreeSlotsCount() > 0) then
                player:delCurrency("imperial_standing", price);
                player:addItem(item);
                player:messageSpecial(ITEM_OBTAINED,item);
                player:setLocalVar("sanctionItem", 0)
            else
                player:messageSpecial(ITEM_CANNOT_BE_OBTAINED,item);
            end
        end
    end
end;

Reply:
> actually just double check they are able to get the item in finish, because the event update can be cheated before they reach here,

as in..before doing addItem, re-check they have the required points and rank etc. because the item selection can be cheated before reaching it.